OpenRA.Editor/Form1.cs

```diff
@@ -318,6 +318,107 @@ void PopulateActorOwnerChooser()
 			surface1.NewActorOwner = ((PlayerReference)actorOwnerChooser.SelectedItem).Name;
 		}
 
+		void SetupMap(Map map, int width, int height, int left, int right, int top, int bottom, bool resize = false)
+		{
+			var badValue = false;
+
+			if (width < 10 || width > 550 || height < 10 || height > 550)
+			{
+				DialogResult warnResult;
+
+				warnResult = MessageBox.Show("This map size is not recommended and can cause problems!\n\nWould you like to use this size anyway?",
+					"OpenRA Editor", MessageBoxButtons.YesNo, MessageBoxIcon.Warning);
```

Comment:
The pulll request states that you check for valid map sizes. Instead this is a check for recommended map sizes if you have performance problems in mind. I had the same idea before. Note that this was rejected in https://github.com/OpenRA/OpenRA/pull/6874 by @pchote.